src/compiler/sys/config.ts

```diff
@@ -3,8 +3,8 @@ import { createLogger } from './logger/console-logger';
 import { createSystem } from './stencil-sys';
 import { setPlatformPath } from '../sys/modules/path';
 
-export const getConfig = (userConfig: d.Config) => {
-  const config = { ...userConfig };
+export const getConfig = (userConfig: d.Config): d.InternalStrictConfig => {
```

Comment:
I can see us doing that in the future.  I didn't do that here only because I felt like this changelist was getting a little large, and wanted to consider that paths that get take to call `getConfig`, `validateConfig`, `loadConfig`, etc.  I think that coalescing these functions might also be partially guided by us making additional fields non-optional, and didn't want to jump the gun too much here.  Thoughts on us (temporarily) punting on that while we get a few more fields made non-optional?

Reply:
sure, I think that makes sense. For instance, if we make `logger` non-optional then we'll have to move code for that into `validateConfig`, so that's a natural point to look at redoing some of this stuff.